diff --git a/dist/types/client/types.d.ts b/dist/types/client/types.d.ts
index c2d694a38aa2d4009c8a7423004aa1ba5da03839..5b75e78372862c9fd20b9f30f841d07341081157 100644
--- a/dist/types/client/types.d.ts
+++ b/dist/types/client/types.d.ts
@@ -41,7 +41,7 @@ type ClientResponseOfEndpoint<T extends Endpoint = Endpoint> = T extends {
     output: infer O;
     outputFormat: infer F;
     status: infer S;
-} ? ClientResponse<O, S extends number ? S : never, F extends ResponseFormat ? F : never> : never;
+} ? O : never;
 export interface ClientResponse<T, U extends number = StatusCode, F extends ResponseFormat = ResponseFormat> extends globalThis.Response {
     readonly body: ReadableStream | null;
     readonly bodyUsed: boolean;
