diff --git a/src/node-haste/DependencyGraph.js b/src/node-haste/DependencyGraph.js
index 7fda8516588a691e4fc1a81dcd38b72e715d95ec..5d450b470424b8b8bcda6ac643f8b19355ccfa83 100644
--- a/src/node-haste/DependencyGraph.js
+++ b/src/node-haste/DependencyGraph.js
@@ -175,10 +175,13 @@ class DependencyGraph extends EventEmitter {
       : fs.realpathSync(filename);
     const sha1 = this._fileSystem.getSha1(resolvedPath);
     if (!sha1) {
-      throw new ReferenceError(`SHA-1 for file ${filename} (${resolvedPath}) is not computed.
-         Potential causes:
-           1) You have symlinks in your project - watchman does not follow symlinks.
-           2) Check \`blockList\` in your metro.config.js and make sure it isn't excluding the file path.`);
+      return getFileHash(resolvedPath)
+      function getFileHash(file) {
+        return require('crypto')
+          .createHash('sha1')
+          .update(fs.readFileSync(file))
+          .digest('hex')
+      }
     }
     return sha1;
   }
