diff --git a/package.json b/package.json
index 5c81976514075634ae6569dab1201fe4a1f01834..97742b1857bda94eb63aa9cb2ed63a13e16120d7 100644
--- a/package.json
+++ b/package.json
@@ -2,6 +2,7 @@
   "name": "react-native-reanimated-toasts",
   "version": "0.4.0",
   "description": "Animated, customizable Toasts for React Native",
+  "react-native": "src/index",
   "source": "./src/index.tsx",
   "main": "./lib/commonjs/index.js",
   "types": "./lib/typescript/commonjs/src/index.d.ts",
diff --git a/src/toast.tsx b/src/toast.tsx
index e1e5d9c0078b2e43e8d0c9a60e149adf64f32d64..d2a942cc1530a9fa5b8168e93f1c6efcd69e4c20 100644
--- a/src/toast.tsx
+++ b/src/toast.tsx
@@ -7,8 +7,8 @@ import { useToastContext } from './context';
 import { ToastSwipeHandler } from './gestures';
 import { type ToastProps } from './types';
 import { useColors } from './use-colors';
-import { toastDefaultValues } from 'src/constants';
-import { cn } from 'src/tailwind-utils';
+import { toastDefaultValues } from './constants';
+import { cn } from './tailwind-utils';
 
 export const Toast: React.FC<ToastProps> = ({
   id,
